lib/executors/PreExecutor.js

```diff
@@ -220,6 +221,25 @@ define([
 					}
 				});
 
+				if (has('host-node')) {
+					var files, splice;
+					[ 'suites', 'functionalSuites' ].forEach(function (name) {
+						config[name].forEach(function (pattern, index) {
+							files = globUtil.sync(pattern);
+
+							// Original value is replaced by glob findings only if something was found.
+							// This allow mixing glob patterns with regular paths.
+							if (files.length > 0) {
+								splice = [index, 1];
+								files.forEach(function (filename) {
+									splice.push(filename);
+								});
+								Array.prototype.splice.apply(config[name], splice);
```

Comment:
You’ll either need to use `config[name].slice(0).forEach` above, or wait to splice in changes ’till after you’ve finished running through the list. It’s easiest to explain why by example:

```js
var array = [ 'a', 'b', 'c' ];
array.forEach(function (item, index) {
  console.log(item);
  if (item === 'b') {
    array.splice(index, 1, 'd', 'e', 'f', 'g');
  }
});
```

logs 'a' 'b' 'e' because the 3rd item in the array ('c') becomes 'e', but we wanted to look at 'c'.